src/InputState.h

```diff
@@ -231,6 +241,92 @@ struct AssociatedPhrasesPlain : InputState {
   const std::vector<ChoosingCandidate::Candidate> candidates;
 };
 
+struct ChineseNumber : InputState {
+  ChineseNumber(std::string number, ChineseNumberStyle style)
+      : number(std::move(number)), style(style) {}
+  ChineseNumber(ChineseNumber const& number)
+      : number(number.number), style(number.style) {}
+
+  std::string composingBuffer() const {
+    if (style == ChineseNumberStyle::LOWER) {
+      return "[中文數字] " + number;
+    } else if (style == ChineseNumberStyle::UPPER) {
+      return "[大寫數字] " + number;
+    } else if (style == ChineseNumberStyle::SUZHOU) {
+      return "[蘇州碼] " + number;
+    }
+    return number;
+  }
+
+  std::string number;
+  ChineseNumberStyle style;
+};
+
+static std::string DateMacros[] = {"MACRO@DATE_TODAY_SHORT",
```

Comment:
Hmm, this should be declared without `static`, and the actual definition moved to an new `InputState.cpp`.

Reply:
Fixed in 92e46601d055fb57699128d5d2cb8634a1fef339